addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py

```diff
@@ -851,6 +860,39 @@ def _correct_invoice_tax_amount(self, tree, invoice):
     # IMPORT : helpers
     # -------------------------------------------------------------------------
 
+    def _ubl_parse_attached_document(self, tree):
+        """
+        In UBL, an AttachedDocument file is a wrapper around multiple different UBL files.
+        According to the specifications the original document is stored within the top most
+        Attachment node either as an Attachment/EmbeddedDocumentBinaryObject or (in special cases)
+        a CDATA string stored in Attachment/ExternalReference/Description.
+
+        We must parse this before passing the original file to the decoder to figure out how best
+        to handle it.
+        """
+        attachment_node = tree.find('{*}Attachment')
+        if attachment_node is None:
+            return tree
+
+        attachment_binary_data = attachment_node.find('./{*}EmbeddedDocumentBinaryObject')
+        if attachment_binary_data is not None \
+                and attachment_binary_data.attrib.get('encodingCode', 'base64') == 'base64' \
+                and attachment_binary_data.attrib.get('mimeCode') in ('application/xml', 'text/xml'):
```

Comment:
I don't know whether these conditions are necessary. `encodingCode` for example is optional.
https://docs.oasis-open.org/ubl/os-UBL-2.1/mod/summary/reports/UBL-AttachedDocument-2.1.html#UDT-BinaryObject.Type:~:text=type%20is%20text.-,encodingCode,optional,-xsd%3AnormalizedString

Maybe we can just try to decode + parse the contents, and if that doesn't work, fall back on the description?